Cardinal command

specificlanguage opened the pull request:
Adds cardinal command to rotate player to cardinal direction + diagonal. Optional argument to N/S/E/W or up/down

https://github.com/CivClassic/Finale/issues/11

specificlanguage committed:
Cardinal command

Adds cardinal command to rotate player to N/S/E/W or up/down

specificlanguage committed:
Update CardinalCommand.java

added what's actually in the issue because i am a big dumb

Maxopoly commented on src/main/java/com/github/maxopoly/finale/command/CardinalCommand.java:
```diff
@@ -0,0 +1,65 @@
+			case "NORTH": case "0":
+				l.setYaw(180f);
+				break;
+			case "WEST": case "-90": case "270":
```
Why does entering -90 or 270 set it to 90? Same for east.

Maxopoly commented on src/main/java/com/github/maxopoly/finale/command/CardinalCommand.java:
```diff
@@ -0,0 +1,65 @@
+				l.setYaw(90f);
+				break;
+			default:
+				p.sendMessage(ChatColor.WHITE + "Cardinal direction required. You can also use degrees, or 'up' or 'down'. B");
```
We typically color error messages in red

Maxopoly commented on src/main/java/com/github/maxopoly/finale/command/CardinalCommand.java:
```diff
@@ -0,0 +1,65 @@
+
+	@Override
+	public List<String> tabComplete(CommandSender commandSender, String[] strings) {
+		return new LinkedList<>();
```
`return doTabComplete(strings [0], Arrays.asList("NORTH","WEST","EAST","SOUTH", "UP","DOWN"), false);`

Also dont name an array of String `strings`, give it a more descriptive name like `args`

Maxopoly commented on src/main/java/com/github/maxopoly/finale/command/CardinalCommand.java:
```diff
@@ -0,0 +1,65 @@
+		}
+
+		String direction = args[0].toUpperCase();
+		Location l = new Location(p.getWorld(), currLocation.getX(), currLocation.getY(), currLocation.getZ());
```
One letter variables names make for unreadable code, outside of loop counter variables they should be avoided

specificlanguage committed:
fixed my mistakes when i was rushing

usually i try and code well but i was distracted when i was making this lol

specificlanguage committed:
Revert "fixed my mistakes when i was rushing"

This reverts commit 749936f8e3db519be529068d1076ba9c493aba22.

specificlanguage committed:
rotation fixes

rotation arguments are now done using yaw values used in-game, previously they were done by using absolute bearing

Maxopoly commented:
Cool, thank you
